tests/app/views/test_search.py

```diff
@@ -20,356 +20,89 @@ def find_search_summary(res_data):
-            '&datacentreTier=tia-942+tier+1')
-        assert_equal(200, res.status_code)
-        summary = find_search_summary(res.get_data(as_text=True))[0]
-        assert_true(
-            '<span class="search-summary-count">1</span> result found' +
-            ' containing <em>email</em> in' +
-            ' <em>Software as a Service</em>' in summary)
-        assert_true('with a <em>Free option</em>' in summary)
-        assert_true('with a minimum contract period of' in summary)
-        assert_true('an <em>Hour</em>' in summary)
-        assert_true('a <em>Day</em>' in summary)
-        assert_true('with a datacentre tier of' in summary)
-        assert_true('<em>TIA-942 Tier 1</em>' in summary)
-
-    def test_should_ignore_unknown_arguments(self):
-        return_value = self.search_results_multiple_page
-        return_value["services"] = [return_value["services"][0]]
-        return_value["meta"]["total"] = 1
-        self._search_api_client.search_services.return_value = return_value
-
-        res = self.client.get(
-            '/g-cloud/search?q=&lot=saas' +
-            '&minimumContractPeriod=hr&minimumContractPeriod=dy')
-
-        assert_equal(200, res.status_code)
-
-    def test_query_text_is_escaped(self):
-        return_value = self.search_results_multiple_page
-        return_value["services"] = [return_value["services"][0]]
-        return_value["meta"]["total"] = 1
-        self._search_api_client.search_services.return_value = return_value
-
-        res = self.client.get('/g-cloud/search?q=<div>XSS</div>')
-
-        assert_equal(200, res.status_code)
-        summary = find_search_summary(res.get_data(as_text=True))[0]
-        assert_in('&lt;div&gt;XSS&lt;/div&gt;', summary)
-
-    def test_summary_for_unicode_query_keywords(self):
-        return_value = self.search_results_multiple_page
-        return_value["services"] = [return_value["services"][0]]
-        return_value["meta"]["total"] = 1
-        self._search_api_client.search_services.return_value = return_value
-
-        res = self.client.get(u'/g-cloud/search?q=email+\U0001f47e&lot=saas')
-        assert_equal(200, res.status_code)
-        summary = find_search_summary(res.get_data(as_text=True))[0]
-        assert_true(
-            u'<span class="search-summary-count">1</span> result found' +
-            u' containing <em>email \U0001f47e</em> in' +
-            u' <em>Software as a Service</em>' in summary)
+class TestCataloguePage(BaseApplicationTest):
+
+    def _setUp(self, api_client, url=''):
+        base_url = '/marketplace/search/suppliers'
+        url = base_url + url
+
+        self.roles_json = self._get_fixture_data('supplier_roles.json')
+        self.results_json = self._get_fixture_data('supplier_results.json')
+
+        api_client.return_value.get_roles.return_value = self.roles_json
+        api_client.return_value.find_suppliers.return_value = self.results_json
+
+        self.response = self.client.get(url)
```

Comment:
If you factor this request out into a method that returns a response and a page, it'll give you a more flexible interface.  It'll be easier to configure the URL, for example, or to make multiple requests in a test.